src/tasks/minions/buyActivity.ts

```diff
@@ -0,0 +1,53 @@
+import { Bank, toKMB } from 'oldschooljs';
+
+import { Time } from 'e';
+import Buyables from '../../lib/data/buyables/buyables';
+import type { BuyActivityTaskOptions } from '../../lib/types/minions';
+import { itemNameFromID } from '../../lib/util';
+import { handleTripFinish } from '../../lib/util/handleTripFinish';
+import { updateBankSetting } from '../../lib/util/updateBankSetting';
+
+export const buyTask: MinionTask = {
+	type: 'Buy',
+	async run(data: BuyActivityTaskOptions) {
+		const { userID, channelID, itemID, quantity } = data;
+		const user = await mUserFetch(userID);
+
+		const buyable = Buyables.find(b => {
+			if (!b.quantityPerHour || !b.outputItems) return false;
+
+			const bank = typeof b.outputItems === 'function' ? b.outputItems(user) : b.outputItems;
+
+			return bank.has(itemID);
+		});
+
+		if (!buyable) {
+			throw new Error(`No buyable found for item ${itemID}`);
+		}
+
+		const { totalCost } = data;
+		const average = Math.floor(totalCost / quantity);
+
+		const itemsPerHour = (quantity / (data.duration / Time.Minute)) * 60;
```

Comment:
Optional/no change needed, but `calcPerHour` function exists to make this a little easier

```ts
import { calcPerHour } from '@oldschoolgg/toolkit/util';
const itemsPerHour  = calcPerHour(quantity, data.duration);
```

Its up to you in this case